app/src/main/java/com/sdpteam/connectout/map/MapViewModelFactory.java

```diff
@@ -0,0 +1,26 @@
+package com.sdpteam.connectout.map;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.ViewModel;
+import androidx.lifecycle.ViewModelProvider;
+
+/**
+ * Used by ViewModelProvider to instantiate the MapViewModel with custom arguments (here the model)
+ */
+public class MapViewModelFactory implements ViewModelProvider.Factory {
+
+    private final MapModel model;
+
+    public MapViewModelFactory(MapModel model) {
+        this.model = model;
+    }
+
+    @NonNull
+    @Override
+    public <T extends ViewModel> T create(Class<T> modelClass) {
+        if (modelClass.isAssignableFrom(MapViewModel.class)) {
+            return (T) new MapViewModel(model);
+        }
+        throw new IllegalArgumentException("Unknown ViewModel class");
+    }
+}
```

Comment:
I wonder if this is the right approach, as `MapViewModelFactory` might be too much in the context of this application, what is the intended thought behind it ?

Reply:
That factory will be used at 3 different places. Once in the EventActivity, another one in the MapViewFragement and also one in the map of all events (coming in next sprint I guess).
People get afraid of creating new classes, for me this helps modularisation and is more describing than inline implementations.

Reply:
I agree, with the usage of MapViewModelFactory then, especially since it provides flexibility in the ViewModel